Add RepeatedRelationshipReference and provide examples (#35)
neo4j/docs-status-codes: modules/ROOT/pages/notifications/all-notifications.adoc

m|WARNING
|Category
m|GENERIC
|===

> [#_neo_clientnotification_statement_RepeatedRelationshipReference]

Reviewer: ```suggestion
[#_neo_clientnotification_statement_repeatedrelationshipreference]
```

Not sure it matters, but all the rest seems to be like this

=== Neo.ClientNotification.Statement.RepeatedRelationshipReference

.Notification category details
[cols="<1s,<4"]
|===
|Code
m|Neo.ClientNotification.Statement.RepeatedRelationshipReference
|Title
a|The query returns no results because a relationship variable is bound more than once.
|Severity
m|WARNING
|Category
m|GENERIC
|===

> 

Reviewer: Maybe we could add an example where we can give a suggestion for improvement, like this maybe:

```suggestion
Query::
+
[source,cypher]
----
MATCH (:A)-[r]->(), ()-[r]->(:B) RETURN r
----
Description of the returned code::
A relationship is referenced more than once in the query, which leads to no results because relationships must not occur more than once in each result. (Relationship `r` was repeated)
Suggestions for improvement::
In order to find all relationships which have a start node with label `A` and end node with label `B`, it could be rewritten to the following:
+
[source, cypher, role="noplay"]
----
MATCH (:A)-[r]->(:B) RETURN r
----
```

.Binding a relationship variable more than once.
====
Query::
+
[source,cypher]
----
MATCH ()-[r]->()<-[r]-() RETURN count(*) AS count
----

Description of the returned code::

> A relationship variable is bound more than once, which leads to no results because relationships must not occur more than once in each result.

Reviewer: This should be the exact detail that they would get back in the description field of the same notification. So if we change it here, we also need to update the code. Since the cut is today, I believe it's better to keep it as it is in the code today:

```suggestion
A relationship is referenced more than once in the query, which leads to no results because relationships must not occur more than once in each result. (Relationship `r` was repeated)
```

====

.Binding a variable-length relationship variable more than once.
====
Query::
+
[source,cypher]
----
MATCH ()-[r*]->()<-[r*]-() RETURN count(*) AS count
----

Description of the returned code::

> A variable-length relationship variable is bound more than once, which leads to no results because relationships must not occur more than once in each result.

Reviewer: ```suggestion
A relationship is referenced more than once in the query, which leads to no results because relationships must not occur more than once in each result. (Relationship `r` was repeated)
```

m|WARNING
|Category
m|GENERIC
|===

[#_neo_clientnotification_statement_repeatedrelationshipreference]

> === Neo.ClientNotification.Statement.RepeatedRelationshipReference

Reviewer: ```suggestion
=== Neo.ClientNotification.Statement.RepeatedRelationshipReference (when run on version 5.5 or newer)
```

> === Neo.ClientNotification.Statement.RepeatedRelationshipReference (when run on version 5.5 or newer)

Reviewer 2: ```suggestion
=== Neo.ClientNotification.Statement.RepeatedRelationshipVariableInPattern (when run on version 5.5 or newer)
```

.Notification category details
[cols="<1s,<4"]
|===
|Code

> m|Neo.ClientNotification.Statement.RepeatedRelationshipReference

Reviewer 2: ```suggestion
m|Neo.ClientNotification.Statement.RepeatedRelationshipVariableInPattern

Reviewer 2: As suggested by Valerio.

Reviewer 2: Sorry, I just saw that this is an existing code. So I guess we can't change the code and the description.

|Title
a|The query returns no results because a relationship variable is bound more than once.
|Severity
m|WARNING
|Category
m|GENERIC
|===

.Binding a relationship variable more than once.
====
Query::
+
[source,cypher]
----
MATCH (:A)-[r]->(), ()-[r]->(:B) RETURN r
----
Description of the returned code::

> A relationship is referenced more than once in the query, which leads to no results because relationships must not occur more than once in each result. (Relationship `r` was repeated)

Reviewer 2: ```suggestion
A relationship is referenced more than once in the query, which leads to no results because relationships must not occur more than once in each result (the repeated relationship variable is `r`).
```

Reviewer: This is the existing description, so let's not update it.

Author: Like Linnéa said, this is already in product as of 5.4.

Suggestions for improvement::

> In order to find all relationships which have a start node with label `A` and end node with label `B`, it could be rewritten to the following:

Reviewer 2: Can we simplify this sentence to something like:
```suggestion
Use one pattern to match all relationships that start with a node with the label `A` and end with a node with the label `B`:
```
